File: utils/topgg.py
import aiohttp
import config as cfg


async def has_voted(user_id: int) -> bool:
    """Check if a user has voted on Top.gg within the last 12 hours."""
    if not cfg.TOPGG_TOKEN or not cfg.TOPGG_BOT_ID:
        return False
    url = f"https://top.gg/api/bots/{cfg.TOPGG_BOT_ID}/check?userId={user_id}"
    try:
        async with aiohttp.ClientSession() as session:
            async with session.get(url, headers={"Authorization": cfg.TOPGG_TOKEN}, timeout=aiohttp.ClientTimeout(total=5)) as r:
                if r.status == 200:
                    data = await r.json()
                    return data.get("voted") == 1
    except Exception:
        pass
    return False

File: src/utils/topgg.js
const https  = require('https');
const cfg    = require('../config');

/**
 * Returns true if the user has voted for the bot on Top.gg
 * within the last 12 hours (Top.gg resets vote status after 12hr).
 */
function hasVoted(userId) {
  return new Promise(resolve => {
    if (!cfg.topggToken || !cfg.topggBotId) return resolve(false);
    const req = https.request(
      {
        hostname: 'top.gg',
        path: `/api/bots/${cfg.topggBotId}/check?userId=${userId}`,
        headers: { Authorization: cfg.topggToken },
      },
      res => {
        let raw = '';
        res.on('data', c => (raw += c));
        res.on('end', () => {
          try { resolve(JSON.parse(raw).voted === 1); }
          catch { resolve(false); }
        });
      }
    );
    req.on('error', () => resolve(false));
    req.end();
  });
}

module.exports = { hasVoted };
